diff --git a/experiment/measurer/run_coverage.py b/experiment/measurer/run_coverage.py
index 05fb2a1c..2f5ec67b 100644
--- a/experiment/measurer/run_coverage.py
+++ b/experiment/measurer/run_coverage.py
@@ -32,10 +32,10 @@ EXIT_BUFFER = 15
 RSS_LIMIT_MB = 2048
 
 # Per-unit processing timeout for libfuzzer merge.
-UNIT_TIMEOUT = 10
+UNIT_TIMEOUT = 0.2
 
 # Max time to spend on libfuzzer merge.
-MAX_TOTAL_TIME = experiment_utils.get_snapshot_seconds()
+MAX_TOTAL_TIME = 90
 
 
 def do_coverage_run(  # pylint: disable=too-many-locals
diff --git a/fuzzers/coverage/fuzzer.py b/fuzzers/coverage/fuzzer.py
index bb2fad34..f6cd1e77 100644
--- a/fuzzers/coverage/fuzzer.py
+++ b/fuzzers/coverage/fuzzer.py
@@ -21,7 +21,7 @@ from fuzzers import utils
 def build():
     """Build benchmark."""
     cflags = [
-        '-fprofile-instr-generate', '-fcoverage-mapping', '-gline-tables-only'
+        '-fprofile-instr-generate', '-fcoverage-mapping', '-gline-tables-only', '-DFRCOV'
     ]
     utils.append_flags('CFLAGS', cflags)
     utils.append_flags('CXXFLAGS', cflags)
@@ -29,5 +29,6 @@ def build():
     os.environ['CC'] = 'clang'
     os.environ['CXX'] = 'clang++'
     os.environ['FUZZER_LIB'] = '/usr/lib/libFuzzer.a'
+    os.environ['FR_COV_BUILD'] = '1'
 
     utils.build_benchmark()
